spec/models/rehearsal_spec.rb

```diff
@@ -19,10 +19,10 @@
   end
 
   it 'can be read by the user to whom this rehearsal belongs' do
-    # rubocop:disable Lint/UselessAssignment
-    rehearsal = Rehearsal.where(user_id = normal_user.id.to_s)
-    # rubocop:enable Lint/UselessAssignment
-    expect(rehearsal).to exist
+    FactoryBot.create(:user, id: 50)
+    rehearsal = FactoryBot.create(:rehearsal, user_id: 50)
+
+    expect(Rehearsal.where(user_id: 50)).to include(rehearsal)
```

Comment:
This is a place where it DOES make sense to assign an ID, no?

Because the whole question of this test is whether the `rehearsal` has the correct `user_id`.

Reply:
Leaving the premise of this spec aside (I think you're testing Rails core functionality here), here is what I would do:
```
    user = FactoryBot.create(:user)
    rehearsal = FactoryBot.create(:rehearsal, user: user)
    expect(Rehearsal.where(user_id: user.id)).to include(rehearsal)
```

Reply:
OK Roger. Changed it to create the user instead of the user ID. 

Whether it tests Rails functionality... 

I want to test to make sure that this PARTICULAR user can access his rehearsal rather than someone else accessing his rehearsal. (I test that in the following spec). I think it makes sense to test that. Yes, it's testing Rails core functionality in the sense that it's testing whether the relationship WORKS. But it's also testing whether in MY application, I've SET UP that relationship.

Reply:
I think doing that means testing structure instead of behaviour, which is not a very widespread practice. If you really want to go there, you can use https://github.com/thoughtbot/shoulda-matchers, which would provide you with a variety of structure testing tools that make this more obvious. 

The two examples would then essentially become:

`it { should  belong_to(:user) }`

Which very accurately reflects what the examples here are really testing.

Reply:
(By the way, you're not really testing that in the following spec, as you're never setting up any `Rehearsal`. It's a false positive.)

Reply:
I added a new PR to address this - adding `shoulda-matcher` gem, for example: https://github.com/Henryvw/stoic_penknife/tree/improve_rehearsal_model_spec